File: backend-python/routes/style_analysis.py
"""
Style Analysis Routes
Define URL endpoints for AI style analysis
"""
from flask import Blueprint
from controllers.style_analysis_controller import style_analysis_controller

# Create blueprint
style_analysis_bp = Blueprint('style_analysis', __name__)

@style_analysis_bp.route('/analyze', methods=['POST'])
def analyze_image():
    """POST /api/style/analyze - Analyze clothing image"""
    return style_analysis_controller.analyze_image()

@style_analysis_bp.route('/profile', methods=['POST'])
def generate_profile():
    """POST /api/style/profile - Generate style profile"""
    return style_analysis_controller.generate_profile()

@style_analysis_bp.route('/recommendations/<int:item_id>', methods=['GET'])
def get_recommendations(item_id):
    """GET /api/style/recommendations/:id - Get recommendations for item"""
    return style_analysis_controller.get_recommendations(item_id)

File: backend/src/routes/styleAnalysis.js
/**
 * Style Analysis Routes
 * Defines URL endpoints and maps them to controller actions
 */

const express = require('express');
const multer = require('multer');
const styleAnalysisController = require('../controllers/styleAnalysisController');

const router = express.Router();

// Configure multer for image uploads
const storage = multer.memoryStorage();
const upload = multer({
  storage: storage,
  limits: {
    fileSize: 10 * 1024 * 1024 // 10MB limit
  },
  fileFilter: (req, file, cb) => {
    if (file.mimetype.startsWith('image/')) {
      cb(null, true);
    } else {
      cb(new Error('Only image files are allowed'));
    }
  }
});

/**
 * POST /api/style/analyze
 * Analyze a single clothing item image
 */
router.post('/analyze', upload.single('image'), styleAnalysisController.analyzeImage.bind(styleAnalysisController));

/**
 * POST /api/style/profile
 * Generate style profile from wardrobe items
 */
router.post('/profile', styleAnalysisController.generateProfile.bind(styleAnalysisController));

/**
 * POST /api/style/recommendations
 * Get shopping recommendations based on item
 */
router.post('/recommendations', styleAnalysisController.getRecommendations.bind(styleAnalysisController));

module.exports = router;
